src/common/gui/SurgeGUIEditor.cpp

```diff
@@ -2127,11 +2128,35 @@ void SurgeGUIEditor::valueChanged(CControl* control)
       // frame->setModalView(nullptr);
       frame->setDirty();
 
-      synth->storage.getPatch().name = patchName->getText();
-      synth->storage.getPatch().author = patchCreator->getText();
-      synth->storage.getPatch().category = patchCategory->getText();
-      synth->storage.getPatch().comment = patchComment->getText();
-      synth->savePatch();
+      /*
+      ** Don't allow a blank patch
+      */
+      std::string whatIsBlank = "";
+      bool haveBlanks = false;
+
+      if (std::string(patchName->getText()).find_first_not_of( ' ' ) == std::string::npos)
```

Comment:
What about a tab character or something even weirder (from the ASCII range)? I think you should validate that a patch

- the only allowed non-displayable character is a space bar.
- has at least one displayable character.
- is at least one character length.

I think these are fair assumptions for a patch name.

Reply:
I think doing this C-style instead of string class magic functions documents actually best how a patch name is validated and probably performs best too.